interfaces/builtin/docker.go

```diff
@@ -0,0 +1,541 @@
+/{,var/}run/runc/       rw,
+/{,var/}run/runc/**     mrwklix,
+
+# Wide read access to /proc, but somewhat limited writes for now
+@{PROC}/ r,
+@{PROC}/** r,
+@{PROC}/[0-9]*/attr/exec w,
+@{PROC}/sys/net/** w,
+@{PROC}/[0-9]*/cmdline r,
+@{PROC}/[0-9]*/oom_score_adj w,
+
+# Wide read access to /sys
+/sys/** r,
+# Limit cgroup writes a bit
+/sys/fs/cgroup/*/docker/   rw,
+/sys/fs/cgroup/*/docker/** rw,
+
+# Allow tracing ourself (especially the "runc" process we create)
+ptrace (trace) peer=@{profile_name},
+
+# Docker needs a lot of caps, but limits them in the app container
+capability,
+
+# Docker does all kinds of mounts all over the filesystem
+/dev/mapper/control rw,
+/dev/mapper/docker* rw,
+/dev/loop* r,
+/dev/loop[0-9]* w,
+mount,
+umount,
+
+# After doing a pivot_root using <graph-dir>/<container-fs>/.pivot_rootNNNNNN,
+# Docker removes the leftover /.pivot_rootNNNNNN directory (which is now
+# relative to "/" instead of "<graph-dir>/<container-fs>" thanks to pivot_root)
+pivot_root,
+/.pivot_root[0-9]*/ rw,
+
+# file descriptors (/proc/NNN/fd/X)
+/[0-9]* rw,
+# file descriptors in the container show up here due to attach_disconnected
+
+# Docker needs to be able to create and load the profile it applies to containers ("docker-default")
+# XXX We might be able to get rid of this if we generate and load docker-default ourselves and make docker not do it.
+/sbin/apparmor_parser ixr,
+/etc/apparmor.d/cache/ r,
+/etc/apparmor.d/cache/.features r,
+/etc/apparmor.d/cache/docker* rw,
+/etc/apparmor/parser.conf r,
+/etc/apparmor/subdomain.conf r,
+/sys/kernel/security/apparmor/.replace rw,
+
+# We'll want to adjust this to support --security-opts...
+change_profile -> docker-default,
+signal (send) peer=docker-default,
+ptrace (read, trace) peer=docker-default,
+
+# Graph (storage) driver bits
+/dev/shm/aufs.xino rw,
+
+#cf bug 1502785
+/ r,
+`
+
+const dockerConnectedPlugAppArmor = `
+# Description: Allow using Docker. Reserved because this gives
+#  privileged access to the service/host.
+# Usage: reserved
+
+# Obviously need to be able to talk to the daemon
+/{,var/}run/docker.sock rw,
+
+@{PROC}/sys/net/core/somaxconn r,
+`
+
+const dockerPermanentSlotSecComp = `
+# The Docker daemon needs to be able to launch arbitrary processes within
+# containers (whose syscall needs are unknown beforehand)
+
+# Calls the Docker daemon itself requires
+#   /snap/docker/VERSION/bin/docker-runc
+#     "do not inherit the parent's session keyring"
+#     "make session keyring searcheable"
+keyctl
+#   /snap/docker/VERSION/bin/docker-runc
+pivot_root
+
+# This list comes from Docker's default seccomp whitelist (which is applied to
+#   all containers launched unless a custom profile is specified or
+#   "--privileged" is used)
+# https://github.com/docker/docker/blob/v1.12.0/profiles/seccomp/seccomp_default.go#L39-L1879
+# $ grep -C1 'ActAllow' -- profiles/seccomp/seccomp_default.go \
+#     | grep 'Name:' \
+#     | cut -d'"' -f2 \
+#     | sort -u
```

Comment:
Along those lines-- perhaps it would be nice to instead of giving a sorted list, use this outline:
- all syscalls we aren't going to allow (ie commented out) with a comment why
- all syscalls docker daemon itself needs to run
- the syscalls docker daemon expects to give to containers

This will allow for auditing the policy a bit better and gives clues as to where to look in docker deeper inspection.